### PR TITLE
#7443 - EAD...?

### PR DESCRIPTION
**Pull request by mcantelon**
Some EAD fixes.

### CODE REVIEW

File: lib/QubitXmlImport.class.php [hunk 1]
```diff
@@ -855,12 +855,12 @@ private function trackEvent($object, $node)
     }
 
     $id = $node->getAttribute('id');
-    if ($id !== NULL)
+    if (!empty($id))
```
Comment (mistydemeo) [inline lib/QubitXmlImport.class.php:858]: Is this right? I believe `NULL` is actually the return value of `getAttribute()` if the requested attribute isn't present.

File: lib/QubitXmlImport.class.php [hunk 2]
```diff
@@ -855,12 +855,12 @@ private function trackEvent($object, $node)
...
     {
       // The ID value is suffixed with its category, e.g. 384_place
       // This is because `id` is required to be unique within the entire
       // document.
-      $id = preg_replace('_' . $key . '$', '', $id);
+      $id = preg_replace('/_' . $key . '$/', '', $id);
```
Comment (mistydemeo) [inline lib/QubitXmlImport.class.php:863]: You can remove this bit - I reworked this in #72, and it's pretty different now!

File: plugins/sfEadPlugin/modules/sfEadPlugin/templates/indexSuccessBodyDidElement.xml.php [hunk 3]
```diff
@@ -88,7 +88,7 @@
...
+    <unitdate <?php if ($date->getActor() !== NULL || $date->getPlace() !== NULL): ?> <?php echo 'id="'.$date->id.'_event"' ?> <?php endif; ?> <?php if ($date->typeId != QubitTerm::CREATION_ID): ?><?php if ($type = $date->getType()->__toString()): ?><?php echo 'datechar="'.strtolower($type).'" ' ?><?php endif; ?><?php endif; ?><?php if ($startdate = $date->getStartDate()): ?><?php echo 'normal="'?><?php echo Qubit::renderDate($startdate) ?><?php if (0 < strlen($enddate = $date->getEndDate())): ?><?php echo '/'?><?php echo Qubit::renderDate($enddate) ?><?php endif; ?><?php echo '"' ?><?php endif; ?> encodinganalog="<?php echo $ead->getMetadataParameter('unitdate') ?>"><?php echo escape_dc(esc_specialchars(Qubit::renderDateStartEnd($date->getDate(array('cultureFallback' => true)), $date->startDate, $date->endDate))) ?></unitdate>
```
Comment (mistydemeo) [inline plugins/sfEadPlugin/modules/sfEadPlugin/templates/indexSuccessBodyDidElement.xml.php]: I think this clashes with a change made in #72.

File: lib/QubitXmlImport.class.php (hunk 1 above)
Comment (mistydemeo) [inline lib/QubitXmlImport.class.php:858]: Ah nope, I'm wrong - it _is_ an empty string!! Good catch.

**Comment by mistydemeo** [comment]
I've merged #72, so this should get rebased on top.

File: plugins/sfEadPlugin/modules/sfEadPlugin/templates/indexSuccessBodyDidElement.xml.php (hunk 3 above)
Comment (mcantelon) [inline plugins/sfEadPlugin/modules/sfEadPlugin/templates/indexSuccessBodyDidElement.xml.php]: Cool... I'll remove it.

**Comment by mcantelon** [comment]
I think I merged this.

**Pull request closed without merging**